From b8192d28bce632e33d7eef485c7c80a06d74e82b Mon Sep 17 00:00:00 2001
From: Lawrence Mok <lawrence.mok@amlogic.com>
Date: Fri, 28 Feb 2014 13:27:36 -0800
Subject: [PATCH 3569/5965] make kernel less dependent on external sources

Kconfig: source -> isource for new nand Kconfig
Makefile: check if external source directory exists before including it,
    and cleanup commented lines
---
 drivers/amlogic/Kconfig  | 25 +++--------------
 drivers/amlogic/Makefile | 60 ++++++++++++++--------------------------
 2 files changed, 25 insertions(+), 60 deletions(-)

diff --git a/drivers/amlogic/Kconfig b/drivers/amlogic/Kconfig
index 5247fc51749e..4de3a80cc291 100755
--- a/drivers/amlogic/Kconfig
+++ b/drivers/amlogic/Kconfig
@@ -21,17 +21,9 @@ source "drivers/amlogic/pinctrl/Kconfig"
 source "drivers/amlogic/power/Kconfig"
 source "drivers/amlogic/securitykeys/Kconfig"
 source "drivers/amlogic/keymanage/Kconfig"
-#source "drivers/amlogic/extern_io/Kconfig"
-#source "drivers/amlogic/minilvds/Kconfig"
 source "drivers/amlogic/efuse/Kconfig"
 source "drivers/amlogic/smartcard/Kconfig"
-#source "drivers/amlogic/gps/Kconfig"
-#source "drivers/amlogic/modem/Kconfig"
-#source "drivers/amlogic/debug/Kconfig"
-#source "drivers/amlogic/sensors/Kconfig"
 source "drivers/amlogic/boot_monitor/Kconfig"
-#source "drivers/amlogic/aml_watchdog/Kconfig"
-#source "drivers/amlogic/spi/Kconfig"
 source "drivers/amlogic/thermal/Kconfig"
 
 #
@@ -42,9 +34,9 @@ source "drivers/amlogic/usb/Kconfig"
 #source "drivers/amlogic/cardreader/Kconfig"
 source "drivers/amlogic/mmc/Kconfig"
 source "drivers/amlogic/spi_nor/Kconfig"
-#source "drivers/amlogic/nand/Kconfig"
-source "../hardware/amlogic/nand/amlnf/Kconfig"
-source "../hardware/amlogic/nand/nand/Kconfig"
+isource "../hardware/amlogic/nand/amlnf/Kconfig"
+isource "../hardware/amlogic/nand/nand/Kconfig"
+
 #
 #	Network devices
 #
@@ -58,10 +50,8 @@ isource "../hardware/wifi/realtek/drivers/8192cu/rtl8xxx_CU/Kconfig"
 isource "../hardware/wifi/realtek/drivers/8192du/rtl8xxx_DU/Kconfig"
 isource "../hardware/wifi/realtek/drivers/8192eu/rtl8192EU/Kconfig"
 source "drivers/amlogic/bluetooth/Kconfig"
-#3G module power manager
-#source "drivers/amlogic/modem/Kconfig"
-#nfc
 source "drivers/amlogic/nfc/Kconfig"
+
 #
 #  Audio
 #
@@ -78,21 +68,14 @@ source "drivers/amlogic/canvas/Kconfig"
 source "drivers/amlogic/display/Kconfig"
 source "drivers/amlogic/hdmi/Kconfig"
 isource "../hardware/tvin/Kconfig"
-#source "drivers/amlogic/cmem/Kconfig"
-#source "drivers/amlogic/jpeglogo/Kconfig"
-#source "drivers/amlogic/dvb/Kconfig"
 source "drivers/amlogic/ppmgr/Kconfig"
 source "drivers/amlogic/camera/Kconfig"
 source "drivers/amlogic/video_dev/Kconfig"
 source "drivers/amlogic/ionvideo/Kconfig"
 source "drivers/amlogic/deinterlace/Kconfig"
-#source "drivers/amlogic/freescale/Kconfig"
 source "drivers/amlogic/mipi/Kconfig"
-#source "drivers/amlogic/d2d3/Kconfig"
 source "drivers/amlogic/amvecm/Kconfig"
 source "drivers/amlogic/dvb_tv/Kconfig"
-#deferred module for speed
-#source "drivers/amlogic/load_deferred_module/Kconfig"
 
 #
 #	GPU
diff --git a/drivers/amlogic/Makefile b/drivers/amlogic/Makefile
index ffb54d919012..e7b0d5247046 100755
--- a/drivers/amlogic/Makefile
+++ b/drivers/amlogic/Makefile
@@ -33,18 +33,11 @@ obj-y += cpufreq/
 obj-$(CONFIG_GPIO_AMLOGIC) += gpio/
 obj-y += pinctrl/
 
-#ifeq ($(CONFIG_AML_NEXT_GEN_NAND),y)
-#ifeq  ($(CONFIG_ARCH_MESON8),y)
-#obj-$(CONFIG_AML_NEXT_GEN_NAND) += amlnf/
-#else
-#obj-$(CONFIG_AM_NAND) += nand/
-#endif
-#ifeq ($(CONFIG_AML_NEXT_GEN_NAND),y)
-obj-$(CONFIG_AML_NEXT_GEN_NAND) += ../../../hardware/amlogic/nand/amlnf/
-#obj-$(CONFIG_AML_NEXT_NAND) += ../../../hardware/amlogic/nand/nand/aml_nftl_new/
-#else
-obj-$(CONFIG_AM_NAND) += ../../../hardware/amlogic/nand/nand/
-#endif
+ifneq ($(wildcard $(srctree)/../hardware/amlogic/nand),)
+	obj-$(CONFIG_AML_NEXT_GEN_NAND) += ../../../hardware/amlogic/nand/amlnf/
+	obj-$(CONFIG_AM_NAND) += ../../../hardware/amlogic/nand/nand/
+endif
+
 obj-y += ethernet/
 
 obj-$(CONFIG_AM_CANVAS) += canvas/
@@ -72,14 +65,12 @@ obj-$(CONFIG_AML_AUDIO_DSP) += audiodsp/
 
 obj-$(CONFIG_AML_RTC)+= rtc/
 
-#obj-$(CONFIG_GPS_POWERCTRL) += gps/
-
 obj-y += power/
 
-#obj-$(CONFIG_AM_GPIO) +=gpio/
-
 obj-y += tvin_v4l2/
-obj-$(CONFIG_TVIN) += ../../../hardware/tvin/
+ifneq ($(wildcard $(srctree)/../hardware/tvin),)
+	obj-$(CONFIG_TVIN) += ../../../hardware/tvin/
+endif
 
 obj-y += thermal/
 
@@ -96,36 +87,27 @@ obj-$(CONFIG_AM_DVB) += dvb_tv/
 obj-$(CONFIG_AM_SMARTCARD) += smartcard/
 
 obj-$(CONFIG_AM_WIFI) += wifi/
-obj-$(CONFIG_SDIO_DHD_CDC_WIFI_40181_MODULE) += ../../../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/
-obj-$(CONFIG_RTL8188EU) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU/
-obj-$(CONFIG_RTL8188EU_MP) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU_MP/
-obj-$(CONFIG_RTL8192CU) += ../../../hardware/wifi/realtek/drivers/8192cu/rtl8xxx_CU/
-obj-$(CONFIG_RTL8192DU) += ../../../hardware/wifi/realtek/drivers/8192du/rtl8xxx_DU/
-obj-$(CONFIG_RTL8192EU) += ../../../hardware/wifi/realtek/drivers/8192eu/rtl8192EU/
+ifneq ($(wildcard $(srctree)/../hardware/amlogic/wifi),)
+	obj-$(CONFIG_SDIO_DHD_CDC_WIFI_40181_MODULE) += ../../../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/
+	obj-$(CONFIG_RTL8188EU) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU/
+	obj-$(CONFIG_RTL8188EU_MP) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU_MP/
+	obj-$(CONFIG_RTL8192CU) += ../../../hardware/wifi/realtek/drivers/8192cu/rtl8xxx_CU/
+	obj-$(CONFIG_RTL8192DU) += ../../../hardware/wifi/realtek/drivers/8192du/rtl8xxx_DU/
+	obj-$(CONFIG_RTL8192EU) += ../../../hardware/wifi/realtek/drivers/8192eu/rtl8192EU/
+endif
 obj-y += bluetooth/
 obj-$(CONFIG_MESON_NFC) += nfc/
-#obj-y += camera/
-#obj-$(CONFIG_AMLOGIC_MODEM) += modem/
-#obj-$(CONFIG_AM_DEBUG) += debug/
-
-obj-$(CONFIG_VIDEO_AMLOGIC_CAPTURE)	+= camera/
 
-#obj-$(CONFIG_AM_SENSORS) += sensors/
+obj-$(CONFIG_VIDEO_AMLOGIC_CAPTURE) += camera/
 
 obj-$(CONFIG_AMLOGIC_MIPI) += mipi/
 
-#obj-$(CONFIG_FREE_SCALE) += freescale/
-
-#obj-$(CONFIG_D2D3_PROCESS) += d2d3/
-
 obj-$(CONFIG_AML_REG_DEBUG) += regd/
 
-#obj-$(CONFIG_AMLOGIC_SPI_HW_MASTER) += spi/
-
-#obj-$(CONFIG_DEFERRED_DRIVERINIT) +=  load_deferred_module/
-
-obj-$(CONFIG_UMP) += ../../../hardware/arm/gpu/ump/
-obj-$(CONFIG_MALI400) += ../../../hardware/arm/gpu/mali/
+ifneq ($(wildcard $(srctree)/../hardware/arm/gpu),)
+	obj-$(CONFIG_UMP) += ../../../hardware/arm/gpu/ump/
+	obj-$(CONFIG_MALI400) += ../../../hardware/arm/gpu/mali/
+endif
 
 obj-$(CONFIG_AMLOGIC_ION) += ion_dev/
 
-- 
2.19.0

